decentraland/screens/EthereumSignUpScreen.js

```diff
@@ -13,31 +13,41 @@ import {
 import { Account } from "tasit-sdk";
 
 export class EthereumSignUpScreen extends React.Component {
-  // TODO: Use Gnosis Safe
   _onboarding = async () => {
     try {
-      const { setAccount } = this.props;
+      const { setAccount, setSetupInProgress } = this.props;
 
       // Note: The timeout for account creation is about ~10 secs.
       // See more: https://github.com/tasitlabs/tasit/issues/42
       const account = Account.create();
       setAccount(account);
 
       const { address: accountAddress } = account;
-      await fundAccount(accountAddress);
 
-      await approveManaSpending(account);
-      showInfo(`Account created and funded!`);
+      const onFundSuccess = async () => {
+        const onApprovalSuccess = () => {
+          setSetupInProgress(false);
+          showInfo(`Account created and funded!`);
+        };
+
+        approveManaSpending(account, onApprovalSuccess);
```

Comment:
I'm wondering whether the await style still makes sense for higher-level functions that wrap Ethereum interactions. I think it doesn't but we'll want to set best practices for the right pattern when using this style to not end up in nested "callback hell"

The callback style of listening for success with a subscription/callback makes the most sense when used along with optimistic updates, which we aren't doing here yet.

Larger topic - probably worth a discussion outside of this thread. This is already a major improvement as is! Nice work!

Reply:
The thing that made me think about this was the two separate steps of funding the account and then approving  mana spending. I think we may want to have a single callback that awaits the success of both operations with a promise.all style?

Reply:
A) Yeah, I think we should refactor the blockchain interactions on the app to establish a pattern / good practices soon (I've extracted your comment to the issue https://github.com/tasitlabs/tasit/issues/151)
B) Yes, make sense. I'll do that.

Reply:
B) Done.

Reply:
Great, thanks. We can continue the rest of the discussion over in that issue.

Reply:
Resolving this